php/WP_CLI/AutoloadSplitter.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WP_CLI;
+
+/**
+ * Class AutoloadSplitter.
+ *
+ * This class is used to provide the splitting logic to the
+ * `wp-cli/autoload-splitter` Composer plugin.
+ *
+ * @package WP_CLI
+ */
+class AutoloadSplitter {
+
+	/**
+	 * Check whether the current class should be split out into a separate
+	 * autoloader.
+	 *
+	 * Note: `class` in this context refers to all PHP autoloadable elements:
+	 *    - classes
+	 *    - interfaces
+	 *    - traits
+	 *
+	 * @param string $class Fully qualified name of the current class.
+	 * @param string $code  Path to the code file that declares the class.
+	 *
+	 * @return bool Whether to split out the class into a separate autoloader.
+	 */
+	public function __invoke( $class, $code ) {
+		return 1 === preg_match( '/.*\/wp-cli\/\w*-command\/.*/', $code )
+			|| 1 === preg_match( '/.*\/php\/commands\/src\/.*/', $code );
```

Comment:
I'm not sure what you mean by that.

The autoloader splitter is meant to separate framework code and bundled commands code (because these are both part of the same `composer.json` file).

Commands pulled-in as additional external commands via the Package Index (or via any other means) are already located in a different autoloader anyway.